src/logging.rs

```diff
@@ -0,0 +1,237 @@
+use std::collections::HashMap;
+use std::env;
+use std::fs;
+use std::num::NonZeroU64;
+use std::sync::Arc;
+use std::sync::OnceLock;
+
+use poise::serenity_prelude::futures::executor::block_on;
+use poise::serenity_prelude::ChannelId;
+use rolling_file::RollingConditionBasic;
+use sqlx::MySql;
+use sqlx::Pool;
+use tracing::error;
+use tracing::info;
+use tracing::Level;
+use tracing::Subscriber;
+use tracing_appender::non_blocking::WorkerGuard;
+use tracing_subscriber::filter;
+use tracing_subscriber::fmt;
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::reload;
+use tracing_subscriber::reload::Handle;
+use tracing_subscriber::Registry;
+
+use crate::bot;
+use crate::mysql_lib;
+
+const LOG_FILE_MAX_SIZE_MB: u64 = 10;
+const MAX_AMOUNT_LOG_FILES: usize = 10;
+
+static DISCORD_LAYER_CHANGE_HANDLE: OnceLock<Handle<DiscordTracingLayer, Registry>> =
+    OnceLock::new();
+
+/// Returns a WorkerGuard which needs to be dropped at the end of the main function,
+/// to ensure the log files get closed.
+pub async fn setup_logging() -> WorkerGuard {
+    // create log dir
+    fs::create_dir_all("./appdata/logs").expect("Could not create log directory");
+    let absolute_path = fs::canonicalize("./appdata/logs").unwrap();
+    println!("Absolute path: {absolute_path:?}");
```

Comment:
Is this still needed? Looks like a debug call

Reply:
Nope, it isn't. Removed it